src/EurobusinessHelper.UI.ASP/app/src/Pages/GamePage/CurrentGame/CurrentGame.tsx

```diff
@@ -27,37 +28,67 @@ interface CurrentGameProps {
 export const CurrentGame = ({ gameId }: CurrentGameProps) => {
   const [amount, setAmount] = useState<string>("");
   const [errorMessage, setErrorMessage] = useState<string>("");
+  const [operationLog, setOperationLog] = useState<string[]>([]);
   const [, setHub] = useState<GameHub | undefined>(undefined);
 
   const gameDetails = useSelector(selectGameDetails(gameId));
   const identity = useAppSelector(selectIdentity);
 
   const dispatch = useDispatch();
   const currentAccountId = useMemo(
-    () =>
-      gameDetails?.accounts?.find((a) => a.email === identity?.email)?.id || "",
+    () => gameDetails?.accounts?.find((a) => a.email === identity?.email)?.id || "",
     [gameDetails, identity]
   );
 
+  const getAccountNameAndEmail = useCallback(
+    (accountId: string) => {
+      const account = gameDetails?.accounts?.find((a) => a.id === accountId);
+
+      return `${account?.name} (${account?.email})`;
+    },
+    [gameDetails]
+  );
+
+  const setLogMessage = useCallback(
+    (logType: GameOperatingLog, toAccount: string, amount: number, fromAccount: string) => {
+      const time = new Date().toLocaleTimeString();
+      let logMessage = "";
+
+      if (logType === GameOperatingLog.TransferCompleted) {
+        logMessage = `[${time}] ${getAccountNameAndEmail(
+          toAccount
+        )} recieved $${amount} from ${getAccountNameAndEmail(fromAccount)}.`;
+      } else if (logType === GameOperatingLog.BankTransferCompleted) {
+        logMessage = `[${time}] ${getAccountNameAndEmail(
+          toAccount
+        )} recieved $${amount} from the bank.`;
+      }
+      setOperationLog([logMessage, ...operationLog]);
+    },
+    [getAccountNameAndEmail, operationLog]
+  );
+
   useEffect(() => {
     if (!currentAccountId) return;
     const hub = new GameHub(
       () => dispatch(fetchDetails(gameId, true)),
       (requestId, accountTo, amount) => {
         if (
           window.confirm(
-            `Account ${accountTo} requested $${amount}. Request id: ${requestId}`
+            `Account ${getAccountNameAndEmail(accountTo)} requested $${amount} from the bank.`
           )
         )
           approveRequest(requestId);
       },
       (account, amount) => {
-        if (window.confirm(`Account ${account} requested $${amount}`))
+        if (window.confirm(`Account ${getAccountNameAndEmail(account)} requested $${amount}.`))
           transferMoney(gameId, currentAccountId, account, amount);
-      }
+      },
+      (logType, toAccount, amount, fromAccount) =>
+        setLogMessage(logType, toAccount, amount, fromAccount)
     );
     hub.initializeAccount(currentAccountId).then(() => setHub(hub));
-  }, [currentAccountId, dispatch, gameId]);
+  }, [currentAccountId, dispatch, gameId, getAccountNameAndEmail, operationLog, setLogMessage]);
```

Comment:
We can't have these dependencies here. This useEffect should be invoked only once as here we create the hub, connect to it and register the account. In this case the hub would be reinitialized every time when the operationLog changes.

To make it work, you'll probably need to extend the hub with methods updating the handlers and then only update the required handler when the dependencies change.

Reply:
The only dependencies here should be the currentAccountId, dispatch, gameId and getAccountNameAndEmail.

If you don't want to refactor the hub, you could also extract the operation log to redux and then you would just dispatch an action with the new log without the dependencies.